src/pepr/operator/controllers/keycloak/authservice/config.ts

```diff
@@ -99,47 +130,112 @@ function buildInitialSecret(): AuthserviceConfig {
   return config;
 }
 
-export async function getAuthserviceConfig() {
-  const authSvcSecret = await K8s(kind.Secret)
+/**
+ * Retrieves the authservice configuration, either from the in-memory cache
+ * or from the Kubernetes secret if not already cached.
+ *
+ * @returns {Promise<AuthserviceConfig>} - The authservice configuration.
+ */
+export async function getAuthserviceConfig(): Promise<AuthserviceConfig> {
+  if (inMemorySecret) {
+    log.info("Returning in-memory authservice secret");
+    return Promise.resolve(inMemorySecret);
+  }
+
+  // Fetch the authservice secret from Kubernetes if not in cache
+  pendingSecretFetch = K8s(kind.Secret)
     .InNamespace(operatorConfig.namespace)
-    .Get(operatorConfig.secretName);
-  return JSON.parse(atob(authSvcSecret!.data!["config.json"])) as AuthserviceConfig;
+    .Get(operatorConfig.secretName)
+    .then(secret => {
+      const config = JSON.parse(atob(secret.data!["config.json"])) as AuthserviceConfig;
```

Comment:
Is it possible that `secret.data` is null? should we have a null check before this `JSON.parse`?

Reply:
I could check and throw an explicit error here but I think I'm ok with it throwing all the way back to the main `handleFailure` in the package reconciler.   I think that would be the only benefit to checking, vs just letting it throw an error but let me know if you see it differently.